src/components/shared/input/index.tsx

```diff
@@ -50,12 +56,23 @@ const Input = <
             }}
             placeholder={placeholder}
             autoCapitalize={autocapitalize}
-            type={type}
+            type={type === 'password' && isPasswordVisible ? 'text' : type}
             InputRightElement={rightElement}
           />
         )}
         name={name}
       />
+      {type === 'password' && (
+        <TouchableOpacity
+          onPress={() => setIsPasswordVisible(!isPasswordVisible)}
+          style={styles.icon}>
+          <MaterialCommunityIcons
+            name={isPasswordVisible ? 'eye' : 'eye-off'}
+            size={28}
+            color="black"
+          />
+        </TouchableOpacity>
+      )}
```

Comment:
NativeBase Input has a prop called `InputRightElement` which seems ideal for this use case, what do you think about moving this component to that prop?